src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3716,40 +3687,60 @@ export class ResultsViewPageStore {
     });
 
     readonly filteredSamples = remoteData({
-        await: () => [
-            this.samples,
-            this.coverageInformation,
-            this.genes,
-            this.selectedMolecularProfiles,
-        ],
+        await: () => [this.samples, this.unprofiledSampleKeyToSample],
         invoke: () => {
             if (this.hideUnprofiledSamples) {
-                // only show samples that are profiled in every gene in every selected profile
-                const genes = this.genes.result!;
-                const coverageInfo = this.coverageInformation.result!;
-                const queryProfileIds = this.selectedMolecularProfiles.result!.map(
-                    p => p.molecularProfileId
-                );
+                const unprofiledSampleKeys = this.unprofiledSampleKeyToSample
+                    .result!;
                 return Promise.resolve(
-                    this.samples.result!.filter(sample => {
-                        return _.every(genes, gene => {
-                            return _.every(
-                                isSampleProfiledInMultiple(
-                                    sample.uniqueSampleKey,
-                                    queryProfileIds,
-                                    coverageInfo,
-                                    gene.hugoGeneSymbol
-                                )
-                            );
-                        });
-                    })
+                    this.samples.result!.filter(
+                        s => !(s.uniqueSampleKey in unprofiledSampleKeys)
+                    )
                 );
             } else {
                 return Promise.resolve(this.samples.result!);
             }
         },
     });
 
+    readonly unprofiledSamples = remoteData({
+        await: () => [
+            this.samples,
+            this.coverageInformation,
+            this.genes,
+            this.selectedMolecularProfiles,
+        ],
+        invoke: () => {
+            const genes = this.genes.result!;
+            const coverageInfo = this.coverageInformation.result!;
+            const queryProfileIds = this.selectedMolecularProfiles.result!.map(
+                p => p.molecularProfileId
+            );
+            return Promise.resolve(
+                this.samples.result!.filter(sample => {
```

Comment:
this condition could use a comment.   like, i don't understand why we use "isSampleProfiledInMultiple" why multiple?

Reply:
also, seems like a very expensive operation, right?   samples x genes x more .   maybe no way around it.

Reply:
`isSampleProfiledInMultiple` is a method that checks whether a sample is profiled in multiple profiles, in a way that's more efficient than checking a single profile multiple times. the logic here (I'll add a comment because it is really hard to read) is we want to only consider a sample unprofiled if it is unprofiled in every gene in every profile. yes, it is expensive - samples x genes x profiles. the worst case is when everything is profiled, otherwise the `_.every` and the `_.some` would both return early.